### ISSUE TITLE
Timeout cannot be set below 890ms

### ISSUE THREAD

**Issue by Gregy** [royalpinto/node-cares#2]
Hi,
first of all thank you for this great library. It helps a lot :)

Now for the bug:

It appears that the timeout option is not entirely respected. I want to set really low value and 1 try but the minimum time I have to wait (with timout=1) is still 890ms. Which can be quite a lot. I first thought that the library is just that slow but it is not the case. If the server responds in for example 20ms then the result is returned immidiately.

Here is proof of concept:

```
'use strict';
const cares = require('cares')

let server = '192.168.2.44'; //nonexistant

const opts = {
  servers: [server],
  timeout: 100,
  tries: 1,
};

const resolver = new cares.Resolver(opts);
const starttime = process.hrtime();
resolver.query("testdomain.test", (err, response) => {
  const endtime = process.hrtime(starttime);
  let endms = ""+Math.floor(endtime[1]/1000000);
  while(endms.length < 3) {
    endms = "0"+endms;
  }
  if(err) {
    console.log(`Query finished in ${endtime[0]}.${endms}s with ${err}`);
  }
  else {
    console.log(`Query finished in ${endtime[0]}.${endms}s with ${response}`);
  }
});
```

I set non existant server to test the timeout. With this code I get:

```
Query finished in 0.890s with Error: Timeout while contacting DNS servers
Query finished in 0.892s with Error: Timeout while contacting DNS servers
Query finished in 0.882s with Error: Timeout while contacting DNS servers
....
```

Thank you

**Comment by Gregy** [comment]
Some more experimentation yielded something bizarre:

```
Query with timeout 50ms finished in 0.894s with Error: Timeout while contacting DNS servers
Query with timeout 100ms finished in 0.906s with Error: Timeout while contacting DNS servers
Query with timeout 150ms finished in 0.906s with Error: Timeout while contacting DNS servers
Query with timeout 200ms finished in 0.906s with Error: Timeout while contacting DNS servers
Query with timeout 250ms finished in 0.906s with Error: Timeout while contacting DNS servers
Query with timeout 300ms finished in 0.906s with Error: Timeout while contacting DNS servers
Query with timeout 350ms finished in 0.905s with Error: Timeout while contacting DNS servers
Query with timeout 400ms finished in 0.905s with Error: Timeout while contacting DNS servers
Query with timeout 450ms finished in 0.905s with Error: Timeout while contacting DNS servers
Query with timeout 500ms finished in 0.905s with Error: Timeout while contacting DNS servers
Query with timeout 550ms finished in 0.905s with Error: Timeout while contacting DNS servers
Query with timeout 600ms finished in 0.905s with Error: Timeout while contacting DNS servers
Query with timeout 650ms finished in 0.904s with Error: Timeout while contacting DNS servers
Query with timeout 700ms finished in 0.904s with Error: Timeout while contacting DNS servers
Query with timeout 750ms finished in 0.904s with Error: Timeout while contacting DNS servers
Query with timeout 800ms finished in 0.904s with Error: Timeout while contacting DNS servers
Query with timeout 850ms finished in 0.903s with Error: Timeout while contacting DNS servers
Query with timeout 900ms finished in 0.903s with Error: Timeout while contacting DNS servers
Query with timeout 950ms finished in 0.903s with Error: Timeout while contacting DNS servers
Query with timeout 1000ms finished in 0.903s with Error: Timeout while contacting DNS servers
Query with timeout 1050ms finished in 0.902s with Error: Timeout while contacting DNS servers
Query with timeout 1100ms finished in 0.902s with Error: Timeout while contacting DNS servers
Query with timeout 1300ms finished in 0.901s with Error: Timeout while contacting DNS servers
Query with timeout 1150ms finished in 1.886s with Error: Timeout while contacting DNS servers
Query with timeout 1200ms finished in 1.886s with Error: Timeout while contacting DNS servers
Query with timeout 1250ms finished in 1.886s with Error: Timeout while contacting DNS servers
Query with timeout 1350ms finished in 1.886s with Error: Timeout while contacting DNS servers
Query with timeout 1400ms finished in 1.886s with Error: Timeout while contacting DNS servers
Query with timeout 1450ms finished in 1.886s with Error: Timeout while contacting DNS servers
```

code:

```
'use strict';
  const cares = require('cares')

  let server = '192.168.2.44'; //nonexistant

  for(let timeout=50; timeout< 1500; timeout+=50) {
    const opts = {
      servers: [server],
      timeout: timeout,
      tries: 1,
    };
    const resolver = new cares.Resolver(opts);
    const starttime = process.hrtime();
    resolver.query("testdomain.test", (err, response) => {
      const endtime = process.hrtime(starttime);
      let endms = ""+Math.floor(endtime[1]/1000000);
      while(endms.length < 3) {
        endms = "0"+endms;
      }
      if(err) {
        console.log(`Query with timeout ${timeout}ms finished in ${endtime[0]}.${endms}s with ${err}`);
      }
      else {
        console.log(`Query with timeout ${timeout}ms finished in ${endtime[0]}.${endms}s with ${response}`);
      }
    });
  }
```

**Comment by royalpinto** [comment]
Hi @Gregy, Thanks for the interest showed in the this module. I have fixed this issue and is available on the latest version `1.1.3`
The issue was that, module was checking for the query timeouts every second using `uv_timer_` and reducing this interval fixed the issue.

**Comment by Gregy** [comment]
Great. Thank you very much. :)
